docs/performance/sharding.md

```diff
@@ -1,41 +1,133 @@
 (sharding-guide)=
-
 (sharding-performance)=
 
-# Sharding Performance Guide
+# Sharding performance guide
 
-This document is a sharding best practice guide for CrateDB.
+:::{div} sd-text-muted
+Applying sharding can drastically improve the performance on large datasets.
+:::
 
-A brief recap: CrateDB tables are split into a configured number of shards, and
-then these shards are distributed across the cluster.
+This document is a sharding best practice guide for CrateDB.
+A brief recap: CrateDB tables are split into a configured number of shards.
+These shards are distributed across the cluster to optimize concurrent and
+parallel data processing.
 
-Figuring out how many shards to use for your tables requires you to think about
-the type of data you're processing, the types of queries you're running, and
-the type of hardware you're using.
+Whenever possible, CrateDB will parallelize query workloads and distribute them
+across the whole cluster. The more CPUs this query workload can be distributed
+across, the faster the query will run.
 
-:::{NOTE}
+:::{seealso}
 This guide assumes you know the basics.
+If you are looking for an intro to sharding, see also the
+{ref}`sharding-partitioning` and the
+{ref}`sharding reference <crate-reference:ddl-sharding>` documentation.
+:::
+
+
+## General recommendations
+
+To avoid running your clusters with too many shards or too large shards,
+implement the following guidelines as a rule of thumb:
+
+- Use shard sizes between 10GB and 50GB.
+
+- Keep the number of records on each shard below 200 million.
+
+
+## Sizing considerations
+
+General principles requires careful consideration for cluster
+sizing and architecture.
+Keep the following things in mind when building your sharding strategy.
+Each shard incurs overhead in terms of open files, RAM allocation, and CPU cycles
+for maintenance operations.
+
+### Shard size vs. number of shards
 
-If you are looking for an intro to sharding, see {ref}`sharding
-<crate-reference:ddl-sharding>`.
+The optimal approach balances shard count with shard size. Individual shards should
+typically contain 3-70 GB of data, with 10-50 GB being the sweet spot for most
```

Comment:
@seut said:

> We should remove this recommendation about 3-70 GB. 10-50 GB is correct.

See also:
- https://opensearch.org/blog/optimize-opensearch-index-shard-size/
- https://www.elastic.co/docs/deploy-manage/production-guidance/optimize-performance/size-shards#general-sizing-guidelines

Reply:
5b28873a58 brings in relevant adjustments.